### PR TITLE
fix(rt): constrain RFC-3339 timestamps to microsecond precision

### PR DESCRIPTION
**Pull request by aajtodd**
## Issue \#

fixes https://github.com/awslabs/aws-sdk-kotlin/issues/304

## Description of changes

Restrict RFC-3339 (ISO-8601) timestamps to microsecond precision

## Scope

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/time/InstantJVM.kt [hunk 1]
```diff
@@ -40,7 +41,7 @@ actual class Instant(internal val value: jtInstant) : Comparable<Instant> {
      * Encode the [Instant] as a string into the format specified by [TimestampFormat]
      */
     actual fun format(fmt: TimestampFormat): String = when (fmt) {
-        TimestampFormat.ISO_8601 -> ISO_INSTANT.format(value)
+        TimestampFormat.ISO_8601 -> ISO_INSTANT.format(value.truncatedTo(ChronoUnit.MICROS))
```
Comment (kneekey23) [inline runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/time/InstantJVM.kt:44]: is there any downside to always defaulting to microseconds? assuming not but just asking

**Review by kneekey23 — APPROVED** [review]
LGTM

File: runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/time/InstantJVM.kt (hunk 1 above)
Comment (aajtodd) [reply runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/time/InstantJVM.kt:44]: The downside is loss of resolution but there is going to be no way to tell which services support finer grained precision.

Comment (kneekey23) [reply runtime/runtime-core/jvm/src/aws/smithy/kotlin/runtime/time/InstantJVM.kt:44]: I figured there was no way to tell

**Review by ianbotsf — APPROVED** [review]

**Pull request merged**